ytgov/sfa-client
# Messaging Module: PDF Generation Service

# Context - DEPENDS ON UNMERGED PR

Fixes https://github.com/icefoganalytics/sfa-client/issues/1

Add ability to generate PDFs to the repo.
Add appropriate services and test routes that can generate either an approval or rejection letter as a PDF.
Add new routes
- `/api/v2/admin/application-letter/:applicationId/approval` - POST
- `/api/v2/admin/application-letter/:applicationId/rejection` - POST

Depends on:
- https://github.com/ytgov/sfa-client/pull/65

# Screenshots

Approval
![image](https://github.com/ytgov/sfa-client/assets/23045206/c61fa8f1-6ef8-41cd-a1c6-b27588783360)
![image](https://github.com/ytgov/sfa-client/assets/23045206/0d6a1da7-1f10-47b7-984d-091f3b8d3249)

Rejection
![image](https://github.com/ytgov/sfa-client/assets/23045206/5f5b5c39-082b-4560-aeef-df455e1c9726)
![image](https://github.com/ytgov/sfa-client/assets/23045206/fb0af996-baf0-42c4-a663-138d93da6a68)


# Testing Instructions

1. Boot the api and database via `dev up`
2. Boot the front-end via `cd src/web && npm run start`
3. Log in to the app at http://localhost:8080/
4. In a new tab go http://localhost:3000/api/v2/admin/application-letter/250/approval.
5. You will get a copy of the Approval Letter, and the filename will include the last name of the user associated with Application#250.
6. Go to http://localhost:3000/api/v2/admin/application-letter/-1/approval, check that the response is a 404.
7. Go to http://localhost:3000/api/v2/admin/application-letter/250/rejection.
8. You will get a copy of the Approval Letter, and the filename will include the last name of the user associated with Application#250.
9. Go to http://localhost:3000/api/v2/admin/application-letter/-1/rejection, check that the response is 404.
